handlers/handlers_test.go

```diff
@@ -186,11 +250,76 @@ func defaultParams() url.Values {
 	values.Set("offset", "0")
 	values.Set("fromDate", "")
 	values.Set("toDate", "")
-	values.Set("sort", "release_date_desc")
+	values.Set("sort", queryparams.RelDateDesc.BackendString())
 	values.Set("keywords", "")
 	values.Set("query", "")
-	values.Set("release-type", "type-upcoming")
+	values.Set("release-type", queryparams.Upcoming.String())
 	values.Set("highlight", "true")
 
 	return values
 }
+
+func defaultICSParams() url.Values {
+	values := url.Values{}
+	values.Set("limit", "1000")
+	values.Set("toDate", time.Now().AddDate(0, 3, 0).Format(queryparams.DateFormat))
+	values.Set("sort", queryparams.RelDateAsc.BackendString())
+	values.Set("release-type", queryparams.Upcoming.String())
+
+	return values
+}
+
+func TestICalDate(t *testing.T) {
+	Convey("given a set of badly formatted date strings", t, func() {
+		bad := []string{"1st Jan 2020", "21-03-2021", "2021-03-04T12:10:00"}
+		Convey("verify the function simply returns an empty string", func() {
+			for _, d := range bad {
+				icd := iCalDate(context.Background(), d)
+				So(icd, ShouldEqual, "")
+			}
+		})
+	})
+
+	Convey("but given a set of well formatted date strings", t, func() {
+		good := []struct{ date, icalDate string }{
+			{date: "2021-03-04T12:10:00Z", icalDate: "20210304T121000Z"},
+			{date: "2021-03-04T12:10:00.000Z", icalDate: "20210304T121000Z"},
+			//{date: "2021-03-04T12:10:00Z05", icalDate: "20210304T071000Z"},
+		}
+		Convey("verify the function returns the date formatted according to the iCal standard", func() {
+			for _, tc := range good {
+				icd := iCalDate(context.Background(), tc.date)
+				So(icd, ShouldEqual, tc.icalDate)
+			}
+		})
+	})
+}
+
+type printer func(b []byte) (int, error)
+
+func (p printer) Write(b []byte) (int, error) {
+	return p(b)
+}
+
+func TestToICSFile(t *testing.T) {
+	Convey("given a list of resources", t, func() {
+		resources := []sitesearch.Release{{URI: "/release/stuff", Description: sitesearch.ReleaseDescription{Title: "A Release Title", ReleaseDate: "2021-03-04T12:10:00Z"}}}
+		Convey("and a bad printer that fails", func() {
+			printerError := errors.New("this is a bad-printer error")
+			badPrinter := printer(func(b []byte) (int, error) { return 0, printerError })
+			Convey("verify that the function returns the error generated by the bad printer", func() {
```

Comment:
I think it's worth mentioning the function name here

```suggestion
			Convey("then toICSFile returns the error generated by the bad printer", func() {
```